# 02 Trend Across Time

## import modules and dataset

In [1]:
import pandas as pd
import altair as alt
alt.renderers.enable('jupyterlab')

RendererRegistry.enable('jupyterlab')

In [2]:
trials_clean = pd.read_csv('./export/trials_clean.csv')
trials_clean.sample(20)

year  decade  century  tried  deaths            city  \
9050   1599.0    1590     1500      1     NaN      Gelnhausen   
1205   1639.0    1630     1600      1     NaN      Rettenberg   
10115  1669.0    1660     1600      1     NaN             NaN   
6101      NaN    1620     1600      3     2.0       Uberherrn   
5486   1662.0    1660     1600      1     0.0             NaN   
8660   1599.0    1590     1500      1     0.0       Neuchatel   
2409   1335.0    1330     1300      1     1.0     Carcassonne   
4858   1650.0    1650     1600      1     0.0             NaN   
6895   1562.0    1560     1500      3     2.0    Sindelfingen   
10709  1610.0    1610     1600      1     0.0         Haarlem   
601    1504.0    1500     1500      1     NaN  Markt Schwaben   
7742   1684.0    1680     1600      1     NaN        Balingen   
8400   1579.0    1570     1500      1     0.0          Geneva   
744    1590.0    1590     1500      1     NaN       Abensberg   
3298   1619.0    1610     1600      1     NaN             NaN   
3843   1634.0    1630     1600      1     0.0             NaN   
1548   1688.0    1680     1600      1     0.0       Neuotting   
5059   1661.0    1660     1600      1     0.0             NaN   
9268   1634.0    1630     1600      1     NaN        Budingen   
6023      NaN    1610     1600      1     NaN         Erratzu   

              gadm.adm2                   gadm.adm1       gadm.adm0  \
9050          Darmstadt                      Hessen         Germany   
1205      Mittelfranken                      Bayern         Germany   
10115           Lysekil             Vastra Gotaland          Sweden   
6101           Saarland                    Saarland         Germany   
5486       East Lothian                    Scotland  United Kingdom   
8660          Neuchatel                   Neuchatel     Switzerland   
2409               Aude        Languedoc-Roussillon          France   
4858       East Lothian                    Scotland  United Kingdom   
6895          Stuttgart           Baden-Wurttemberg         Germany   
10709           Haarlem               Noord-Holland     Netherlands   
601       Mittelfranken                      Bayern         Germany   
7742           Tubingen           Baden-Wurttemberg         Germany   
8400             Geneve                      Geneve     Switzerland   
744       Mittelfranken                      Bayern         Germany   
3298                NaN                    Scotland  United Kingdom   
3843   Scottish Borders                    Scotland  United Kingdom   
1548      Mittelfranken                      Bayern         Germany   
5059              Angus                    Scotland  United Kingdom   
9268          Darmstadt                      Hessen         Germany   
6023            Navarra  Comunidad Foral de Navarra           Spain   

             lon        lat          record.source  
9050    9.186370  50.204600           Niess (1982)  
1205   10.292200  47.574300       Behringer (1987)  
10115        NaN        NaN          Sörlin (1999)  
6101    6.700610  49.239200          Hiegel (1961)  
5486         NaN        NaN  Goodare et al. (2003)  
8660    6.931930  46.993000          Monter (1976)  
2409    2.353663  43.212161         Carlson (2004)  
4858         NaN        NaN  Goodare et al. (2003)  
6895    9.004410  48.707500       Midelfort (1972)  
10709        NaN        NaN          Waardt (1991)  
601    11.863000  48.190900       Behringer (1987)  
7742    8.849530  48.274800       Midelfort (1972)  
8400    6.142300  46.198400          Monter (1976)  
744    11.850800  48.817700       Behringer (1987)  
3298         NaN        NaN  Goodare et al. (2003)  
3843         NaN        NaN  Goodare et al. (2003)  
1548   12.689600  48.241300       Behringer (1987)  
5059         NaN        NaN  Goodare et al. (2003)  
9268    9.111380  50.286300           Niess (1982)  
6023   -1.456790  43.182300      Henningsen (1980)

### First glance: total casualty
In this dataset:

In [3]:
# how many people are tried in total
# df.sum(axis=0) Summing columns, or summing across rows, set axis=0
trials_clean['tried'].sum(axis=0)

43240

In [4]:
# how many people are killed in total
trials_clean['deaths'].sum(axis=0)

16333.0

So this is just a subset of the people tried and persecuted during the witch trial. According to [Wikipedia's Witch Trials in the early modern period](https://en.wikipedia.org/wiki/Witch_trials_in_the_early_modern_period), between 1400 to 1782, around 40,000 to 60,000 people were killed due to suspicion that they were practicing witchcraft. Some sources estimate that a total of 100,000 trials occurred at its maximum for a similar period.

## Q1: What time period (century/decade/year) witnessed the peak of witch trials?

### Witch Trials by century - line chart

In [5]:
# Group by century 
trials_by_century = trials_clean.groupby([pd.Grouper(key='century')]).count()[['tried']].reset_index()

# rename the columns
trials_by_century.columns = ['century', 'trials_count']
trials_by_century

century  trials_count
0     1300           150
1     1400           607
2     1500          2364
3     1600          7164
4     1700           652
5     1800             3

In [6]:
# line chart by altair
alt.Chart(trials_by_century).mark_line().encode(
    x=alt.X('century'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by century',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


#### The zenith between 1560 and 1630
The period between 1560 and 1630 was known as the 'Great Hunt', when 60% of the prosecution took place. Over the course of a century and a half, 80,000 people were tried for witchcraft and half of them were executed, often burned alive. This time period coincided with the Counter-Reformation and the European wars of religion. [Economists Peter T. Leeson and Jake Russ argued](https://www.peterleeson.com/witch_trials.pdf) that the surgence of European witch trials reflected non-price competition between the Catholic and Protestant churches for religious market share in confessionally contested parts of Christendom.

Simply put, as competing Catholic and Protestant churches vied to win over or retain their followers, they needed to make an impact – and witch trials were the battleground they chose ([the Guardian](https://www.theguardian.com/society/2018/jan/07/witchcraft-economics-reformation-catholic-protestant-market-share)).

### Witch Trials By Decade - line chart

In [7]:
# Group by decade 
trials_by_decade = trials_clean.groupby([pd.Grouper(key='decade')]).count()[['tried']].reset_index()

# rename the columns
trials_by_decade.columns = ['decade', 'trials_count']
trials_by_decade

decade  trials_count
0     1300            11
1     1310            21
2     1320            21
3     1330            25
4     1340             8
5     1350             4
6     1360             3
7     1370            15
8     1380            17
9     1390            25
10    1400            32
11    1410            20
12    1420            35
13    1430            52
14    1440            57
15    1450            90
16    1460            65
17    1470            68
18    1480           101
19    1490            87
20    1500            39
21    1510            43
22    1520            64
23    1530            73
24    1540           106
25    1550           111
26    1560           309
27    1570           291
28    1580           373
29    1590           955
30    1600           621
31    1610           878
32    1620           934
33    1630           724
34    1640          1111
35    1650           872
36    1660          1050
37    1670           419
38    1680           260
39    1690           295
40    1700           195
41    1710           107
42    1720           105
43    1730            68
44    1740            89
45    1750            62
46    1760             8
47    1770            11
48    1780             3
49    1790             4
50    1820             1
51    1830             1
52    1850             1

In [8]:
# line chart by altair
alt.Chart(trials_by_decade).mark_line().encode(
    x=alt.X('decade'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by decade',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


### Witch Trials By Year - line chart

In [9]:
# Group by year 
trials_by_year = trials_clean.groupby([pd.Grouper(key='year')]).count()[['tried']].reset_index()

# rename the columns
trials_by_year.columns = ['year', 'trials_count']

In [10]:
# line chart by altair
alt.Chart(trials_by_year).mark_line().encode(
    x=alt.X('year'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by year',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


According to the data we have, the 16 century witnessed the highest witch trials count. There are 1050 cases in 1660 alone. 